src/pipeline/pipeline-graph.ts

```diff
@@ -42,35 +42,46 @@ export async function calculatePipelineRunGraph(document: VirtualDocument, pipel
     return []; // TODO: throw error there
   }
   let tasks: DeclaredTask[];
-  const uri = document.uri.toString();
-  if (pipelineRunTaskCache.has(uri)) {
-    tasks = [...pipelineRunTaskCache.get(uri)];
-  } else {
-    const refOrSpec = pipelineRunYaml.getTektonPipelineRefOrSpec(doc);
-    if (typeof refOrSpec === 'string') {
-      // get ref pipeline definition
-      const uri = tektonFSUri(ContextType.PIPELINE, refOrSpec, 'yaml');
-      const pipelineDoc = await tektonVfsProvider.loadTektonDocument(uri, false);
-      const pipeDoc = await getPipelineDocument(pipelineDoc, TektonYamlType.Pipeline);
-      tasks = pipelineYaml.getPipelineTasks(pipeDoc);
-
-    } else if (Array.isArray(refOrSpec)) {
-      tasks = refOrSpec;
-    } else {
-      tasks = [];
-    }
+  // const uri = document.uri.toString();
+  // if (pipelineRunTaskCache.has(uri)) {
```

Comment:
can you remove `const pipelineRunTaskCache = new Map<string, DeclaredTask[]>();` global variable if you are not using it.